src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -25,11 +25,303 @@
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serialisedBytes Data from nvm
+    /// @param timeMs transition time in ms
+    /// @return CHIP_NO_ERROR if value as expected, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
+                          scenes::TransitionTimeMs timeMs) override
+    {
+        app::DataModel::DecodableList<Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+
+        ReturnErrorOnFailure(DecodeAttributeValueList(serializedBytes, attributeValueList));
+
+        size_t attributeCount = 0;
+        uint32_t attributeBuffer[kColorControlMaxScenableAttributes];
+        auto pair_iterator = attributeValueList.begin();
+
+        // The color control cluster should have a maximum of 9 scenable attributes
+        ReturnErrorOnFailure(attributeValueList.ComputeSize(&attributeCount));
+        VerifyOrReturnError(attributeCount <= kColorControlMaxScenableAttributes, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+        uint8_t attIdx;
+        while (pair_iterator.Next())
+        {
+            auto & decodePair = pair_iterator.GetValue();
+
+            switch (decodePair.attributeID)
+            {
+            case Attributes::CurrentX::Id:
+                attIdx = ColorControlEFS::kCurrentX;
+                break;
+            case Attributes::CurrentY::Id:
+                attIdx = ColorControlEFS::kCurrentY;
+                break;
+            case Attributes::EnhancedCurrentHue::Id:
+                attIdx = ColorControlEFS::kEnhancedCurrentHue;
+                break;
+            case Attributes::CurrentSaturation::Id:
+                attIdx = ColorControlEFS::kCurrentSaturation;
+                break;
+            case Attributes::ColorLoopActive::Id:
+                attIdx = ColorControlEFS::kColorLoopActive;
+                break;
+            case Attributes::ColorLoopDirection::Id:
+                attIdx = ColorControlEFS::kColorLoopDirection;
+                break;
+            case Attributes::ColorLoopTime::Id:
+                attIdx = ColorControlEFS::kColorLoopTime;
+                break;
+            case Attributes::ColorTemperatureMireds::Id:
+                attIdx = ColorControlEFS::kColorTemperatureMireds;
+                break;
+            case Attributes::EnhancedColorMode::Id:
+                attIdx = ColorControlEFS::kEnhancedColorMode;
+                break;
+            default:
+                return CHIP_ERROR_INVALID_ARGUMENT;
+                break;
+            }
+
+            attributeBuffer[attIdx] = decodePair.attributeValue;
+        }
+
+        ReturnErrorOnFailure(pair_iterator.GetStatus());
+
+        // Switch to the mode saved in the scene
+        if (SupportsColorMode(endpoint, static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kEnhancedColorMode])))
```

Comment:
Please store `static_cast<uint8_t>(attributeBuffer[ColorControlEFS::kEnhancedColorMode])` in a temporary, probably one called `colorMode`, so it's not being copy/pasted.